core/iwasm/common/wasm_runtime_common.c

```diff
@@ -863,6 +863,183 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
     return false;
 }
 
+uint32 parse_args_to_uint32_array(WASMType *type,
+                                uint32 num_args, wasm_val_t *args,
+                                uint32 *out_argv)
+{
+    int i, p;
+
+    for (i = 0, p = 0; i < num_args; i++) {
+        switch (args[i].kind) {
+            case WASM_I32:
+                *(int32 *)out_argv = args[i].of.i32;
+                out_argv++;
+                p++;
+                break;
+            case WASM_I64:
+                *(int64 *)out_argv = args[i].of.i64;
+                out_argv += 2;
+                p += 2;
+                break;
+            case WASM_F32:
+                *(float32 *)out_argv = args[i].of.f32;
+                out_argv++;
+                p++;
+                break;
+            case WASM_F64:
+                *(float64 *)out_argv = args[i].of.f64;
+                out_argv += 2;
+                p += 2;
+                break;
+        }
+    }
+    return p;
```

Comment:
Seems that we can store out_argv as out_argv_org at the beginning, and here
  return out_argv - out_argv_org;
No need to p++ or p += 2 each time